src/tools/sssctl/sssctl_cache.c

```diff
@@ -295,10 +295,12 @@ static const char *sssctl_create_filter(TALLOC_CTX *mem_ctx,
     const char *filter;
     char *filter_value;
     bool qualify_attr = false;
+    bool use_alias = false;
 
     if (strcmp(attr_name, SYSDB_NAME) == 0) {
         if (obj_type == CACHED_USER || obj_type == CACHED_GROUP) {
             qualify_attr = true;
+            use_alias = true;
```

Comment:
Is there a reason to not search by alias always if it exists? I think it would make the logic in the tools simpler in the sense that the back end would store the alias if it's needed (=if the domain is case insensitive or if the user has multiple names).